### PR TITLE
DAP-939: Health services endpoint

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[DAP-939](https://citz-cirmo.atlassian.net/browse/DAP-939)

- Added rabbit health check and endpoint and refactored route of test queue
- Refactored ches health check into a function
- Added `/health/services` endpoint

## 🔰 Checklist

- [x] I have read and agree with the following checklist.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by TaylorFries — DISMISSED** [review]
having one health check for all endpoints is great. works well. tests pass. updates to routing work as expected.

File: backend/src/modules/health/controllers/services.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	const result = getStandardResponse({
+		data,
+		message: allServicesHealthy
+			? "All services are healthy."
+			: "One or more services are unhealthy.",
+		success: allServicesHealthy,
+	});
+
+	res.status(allServicesHealthy ? 200 : 500).json(result);
```
Comment (BradyMitch) [inline backend/src/modules/health/controllers/services.ts:30]:
```suggestion
	res.status(allServicesHealthy ? 200 : 503).json(result);
```

File: backend/tests/modules/health/controllers/health.test.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		(checkS3Connection as jest.Mock).mockResolvedValue(false); // Mock S3 connection as unhealthy
+		(checkRabbitConnection as jest.Mock).mockReturnValue(true); // Mock Rabbit connection as healthy
+		(checkChesHealth as jest.Mock).mockResolvedValue({ success: true }); // Mock CHES health as healthy
+
+		await services(req, res, next);
+
+		expect(checkS3Connection).toHaveBeenCalled();
+		expect(checkRabbitConnection).toHaveBeenCalled();
+		expect(checkChesHealth).toHaveBeenCalled();
+		expect(res.status).toHaveBeenCalledWith(500);
```
Comment (BradyMitch) [inline backend/tests/modules/health/controllers/health.test.ts:100]:
```suggestion
		expect(res.status).toHaveBeenCalledWith(503);
```

File: backend/tests/modules/health/controllers/health.test.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		(checkS3Connection as jest.Mock).mockResolvedValue(true); // Mock S3 connection as healthy
+		(checkRabbitConnection as jest.Mock).mockReturnValue(false); // Mock Rabbit connection as unhealthy
+		(checkChesHealth as jest.Mock).mockResolvedValue({ success: true }); // Mock CHES health as healthy
+
+		await services(req, res, next);
+
+		expect(checkS3Connection).toHaveBeenCalled();
+		expect(checkRabbitConnection).toHaveBeenCalled();
+		expect(checkChesHealth).toHaveBeenCalled();
+		expect(res.status).toHaveBeenCalledWith(500);
```
Comment (BradyMitch) [inline backend/tests/modules/health/controllers/health.test.ts:128]:
```suggestion
		expect(res.status).toHaveBeenCalledWith(503);
```

File: backend/tests/modules/health/controllers/health.test.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+		(checkS3Connection as jest.Mock).mockResolvedValue(true); // Mock S3 connection as healthy
+		(checkRabbitConnection as jest.Mock).mockReturnValue(true); // Mock Rabbit connection as healthy
+		(checkChesHealth as jest.Mock).mockResolvedValue({ success: false }); // Mock CHES health as unhealthy
+
+		await services(req, res, next);
+
+		expect(checkS3Connection).toHaveBeenCalled();
+		expect(checkRabbitConnection).toHaveBeenCalled();
+		expect(checkChesHealth).toHaveBeenCalled();
+		expect(res.status).toHaveBeenCalledWith(500);
```
Comment (BradyMitch) [inline backend/tests/modules/health/controllers/health.test.ts:156]:
```suggestion
		expect(res.status).toHaveBeenCalledWith(503);
```

File: backend/src/modules/rabbit/controllers/health.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	const { getStandardResponse } = req;
+
+	const connected = checkRabbitConnection();
+
+	const result = getStandardResponse({
+		message: `Rabbit connection ${connected ? "successful." : "failed."}`,
+		success: connected,
+	});
+
+	res.status(connected ? 200 : 500).json(result);
```
Comment (BradyMitch) [inline backend/src/modules/rabbit/controllers/health.ts:15]:
```suggestion
	res.status(connected ? 200 : 503).json(result);
```

File: backend/tests/modules/rabbit/controllers/health.test.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		const res = mockResponse();
+		const next = jest.fn();
+		const message = "Rabbit connection failed.";
+
+		(checkRabbitConnection as jest.Mock).mockReturnValue(false); // Mock failed connection
+
+		await health(req, res, next);
+
+		expect(checkRabbitConnection).toHaveBeenCalled();
+		expect(res.status).toHaveBeenCalledWith(500);
```
Comment (BradyMitch) [inline backend/tests/modules/rabbit/controllers/health.test.ts:77]:
```suggestion
		expect(res.status).toHaveBeenCalledWith(503);
```

File: backend/src/modules/s3/controllers/health.ts [hunk 7]
```diff
@@ -5,12 +5,12 @@ import { checkS3Connection } from "../utils";
...
 
 	const result = getStandardResponse({
-		message: `S3 connection ${s3Connected ? "successful." : "failed. Ensure you are on the BC Gov network or VPN."}`,
-		success: s3Connected,
+		message: `S3 connection ${connected ? "successful." : "failed. Ensure you are on the BC Gov network or VPN."}`,
+		success: connected,
 	});
 
-	res.status(s3Connected ? 200 : 500).json(result);
+	res.status(connected ? 200 : 500).json(result);
```
Comment (BradyMitch) [inline backend/src/modules/s3/controllers/health.ts:15]:
```suggestion
	res.status(connected ? 200 : 503).json(result);
```

**Review by TaylorFries — APPROVED** [review]
tests pass, and work as expected, health routers work well, no issues found

**Pull request merged**